authorizerd.go

```diff
@@ -356,13 +356,17 @@ func (a *authority) AuthorizeAccessToken(ctx context.Context, tok, act, res stri
 func (a *authority) authorize(ctx context.Context, m mode, tok, act, res string, cert *x509.Certificate) (Principal, error) {
 	var key string
 
+	if cert != nil {
+		key = cert.Issuer.CommonName + ":" + cert.Subject.CommonName
+	}
+
 	if a.disablePolicyd {
-		key = tok
+		key = key + tok
 	} else {
 		if act == "" || res == "" {
 			return nil, errors.Wrap(ErrInvalidParameters, "empty action / resource")
 		}
-		key = tok + act + res
+		key = key + tok + act + res
 	}
```

Comment:
1. how about using `strings.Builder`?
   - https://stackoverflow.com/questions/1760757/how-to-efficiently-concatenate-strings-in-go/47798475#47798475
2. add delimiter between all sub-strings

---

1. var key strings.Builder
2. key.WriteString(tok); key. WriteRune(':');
3. if cert != nil, key.WriteString(common_name); key. WriteRune(':');
4. if !disablePolicyd, key.WriteString(act); key. WriteRune(':'); key. WriteString(res);

Reply:
Fixed
https://github.com/yahoojapan/athenz-authorizer/pull/75/commits/ccdbdd41c43685852a6d6e66cd1fe1ed9c8e0533